Error when using `sum` on array of floats

Example code:

```py
from seahorse.prelude import *

# This is your program's public key and it will update
# automatically when you build the project.
declare_id('FLwzxcGsWo28gVDA7oGHd15sSnmY6kvP8kziJ6gQqCnC')

@instruction
def ix(sig: Signer):
  a = array(1.0, 1.5)
  s = sum(a)
  print(s)
```

Playground link: https://beta.solpg.io/63b55b8fc6bb8fea17cf0b75

Error:

```
Building...
   error[E0277]: cannot add `f64` to `{integer}`
 --> /src/dot/program.rs:1:557
  |
1 | ...clone ())) . fold (0 , | accum , elem | accum + elem) ; solana_program :: msg ! ("{}" , s) ; }
  |                                                  ^ no implementation for `{integer} + f64`
  |
  = help: the trait `Add<f64>` is not implemented for `{integer}`
```

Error seems to be coming from here: https://github.com/ameliatastic/seahorse-lang/blob/e1178ed959246417255321a4cbe5b55431a75994/src/core/compile/builtin/python.rs#L519-L523

This might be tricky to fix because the comment suggests we don't have type info about the array there

--- a/src/core/compile/builtin/python.rs
+++ b/src/core/compile/builtin/python.rs
@@ -514,12 +514,22 @@ impl BuiltinSource for Python {
                     Transformation::new(|mut expr| {
                         let iterable = match1!(expr.obj, ExpressionObj::Call { args, .. } => args.into_iter().next().unwrap());
 
-                        // TODO need to prevent errors by checking the type here like `Op::Add` does
+                        let init = match &expr.ty {
+                            Ty::Generic(TyName::Builtin(Builtin::Prelude(Prelude::RustInt(..))), _) => quote! { 0 },
+                            Ty::IntParam(..) => quote! { 0 },
+                            Ty::Generic(TyName::Builtin(Builtin::Prelude(Prelude::RustFloat)), _) => quote! { 0.0 },
+                            ty => {
+                                return Err(CoreError::make_raw(
+                                    format!("cannot perform sum of type {}", ty),
+                                    "Hint: sums can be performed on numeric types only."
+                                ))
+                            }
+                        };
 
                         expr.obj = ExpressionObj::Rendered(quote! {
                             // Rolling my own sum implementation here because Rust's `Iterator.sum`
                             // requires type info which is annoying to obtain
-                            #iterable.fold(0, |accum, elem| accum + elem)
+                            #iterable.fold(#init, |accum, elem| accum + elem)
                         });
 
                         Ok(Transformed::Expression(expr))

